core/trino-main/src/main/java/io/trino/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -416,11 +457,16 @@ public Type getFinalType()
         @Override
         public Type getIntermediateType()
         {
-            throw new UnsupportedOperationException();
+            return spillableStateHolder.orElseThrow(UnsupportedOperationException::new).getIntermediateType();
         }
 
         @Override
         public void addInput(GroupByIdBlock groupIdsBlock, Page page)
+        {
+            spillableStateHolder.ifPresentOrElse(holder -> holder.addInput(groupIdsBlock, page), () -> addInputToUnderlyingAccumulator(groupIdsBlock, page));
```

Comment:
Since both lambdas are capturing I wonder if lambda-objects are allocated here? I wonder if it might be better to avoid using lambdas here, since this method could be performance critical? (same for `OrderingGroupedAccumulator`)

Reply:
Done